### PR TITLE
PROGRESS: Dungeons Functionalities

### PR DESCRIPTION
**Pull request by OcelotWalrus**
# PROGRESS/CONTENT

This PR addresses the bug/feature described in issue #89
This PR addresses the game progression goal #'Complete dungeons functionalities'

## Summary

<div type='discussion-op-text'>

<sup>Originally posted by **OcelotWalrus** March 24, 2024 at #89 </sup>
A dungeon will be a new map zone type, which'll be interactive, as hostels and etc...
When you'll run the `Z` command, it'll display a generic menu that have these options:

1. Enter Dungeon
2. See Dungeon Map
3. Exit

- When selecting the option 1, it'll make you enter the dungeon and a new UI will be launched. See below. 
- When selecting the option 2, it'll display a small ASCII art map of the dungeon. This options will not always be there, it'll depends on the dungeon data. (some dungeon won't have any map).

---

About the new UI: the 'dungeon UI':

```yaml
=======================================================
Defy the dungeon <dungeon name>!
=======================================================
PROGRESS: 24% |███_______|
=======================================================
CURRENT ROOM: <room number>
TYPE: <room type>
=======================================================
  - [S]tart Room
  - [C]heck Dungeon Map
  - [I]nventory
  - [P]ause game
  - [E]xit Dungeon
=======================================================

> 

```

The player will start by the room 1, the room starts when he decide. If he succeed, he access to the next room, and the room will start when he decide. Once he complete the dungeon, a dialog will be shown. Exiting the dungeon will abort the process and the player will have to start over. Once the player starts a dungeon, he'll have to finish it, before being able to quit or save the game.

---

And the definition of a dungeon:

```html
Cool Dungeon:
  [normal map zone data]
  dungeon:
    name: <dungeon display name(str)>
    dungeon map: <img name(str)>  # not required
    rooms number: <number of rooms(int)>
    reward dialog: <dialog id(str)>
    no escape: <false | true(bool)>
    rooms:
      <room id>:
        room type: <fight | enigma | boss-fight>  # any ideas?
        room digit: <number of the room(int)>
        room fight data:  # if this is a fight or boss-fight room
          enemy list spawn: <enemy list id(str)>
          no run away: <false | true(bool)>  # let the player flee from battle
          item reward: <items id(list)>  # not required. Maybe other types of reward?
          gold reward: <payment(float)>  # not required. Maybe other types of reward?
        room enigma data:
          script name: <script name.py(str)>
          arguments: <variables id(list)>  # not required
```
</div>

---

This PR also adds many different dungeon enigmas (you can also call that mini-games): one where you gotta reconstitute a sentence whose words have been shuffled, another one where you gotta flip three levers at the right pattern, another one where you gotta guess which enemy is it, from its ASCII art... The PR also adds many more item thumbnails.

## Save File

<details><summary>The following save's fine</summary>

```yaml
active effects:
  a0f96365-693d-4b73-815f-e009282f5496:
    effect duration: 0.7
    effect starting time: 150.8567808616691
    effects:
      critical hit chance coefficient: 1.27
      damage coefficient: 1.27
    type: strength
  e58d6241-bfe6-441c-9d2d-243798fa0ba6:
    effect duration: 0.7
    effect starting time: 150.8567808616691
    effects:
      agility coefficient: 1.15
    type: agility
  e67ae5d5-18a0-4610-9d30-ae53d36bc999:
    effect duration: 0.7
    effect starting time: 150.8567808616691
    effects:
      protection coefficient: 1.25
    type: protection
active missions: []
agility: 2.1275
armor protection: 2.325
attributes:
- None
- seen warg attacking village
- met jerry the junk seller
- has finished jacky deliver
- has mission cut wood and get back home offered
- heard someone shouting in the woods
- has saved the guy in the forlindon woods
- completed goro mounts mines dungeon
cheat: 0
completed dungeons: []
critical hit chance: 0.5715
current mount: 3cd21029-36c6-41d8-8960-d2004b4bdb03
defeated enemies:
- None
- 5
- 60
- 6
- 37
- 101
- 649
- 649
- 520
- 391
- 520
- 391
- 262
- 907
- 907
- 907
- 907
- 6734
- 6602
- 7506
- 7506
- 7377
- 7248
- 7377
- 7506
- 7248
- 7248
- 6988
- 6985
- 6983
- 6982
- 6979
- 6329
- 5684
- 5297
- 4781
- 4523
- 4136
- 4007
- 1165
- 1165
- 6727
- 6726
- 6855
- 6855
- 6984
- 7113
- 7242
- 7242
- 6728
- 3873
- 3874
- 3875
- 3876
- 3875
- 3876
- 3877
- 3877
- 3878
- 3879
- 3875
- 3873
- 3744
- 6987
- 6988
- 6987
- 6986
- 6987
- 7116
- 7113
- 7114
- 7115
- 646
- 6730
- 6992
- 7508
- 2463
- 2464
- 2464
- 5954
- 5954
- 6988
- 6987
- 6987
- 7118
- 7118
- 5164
- 5165
- 5165
- 5165
- 5165
- 6988
difficulty mode: 1
discounts:
  Clarc Warg Stable:
    dropoff: null
    next discount: 57.61452534463683
    remaining time: null
  Dwarf Harbor:
    dropoff: null
    next discount: 21.61452534463673
    remaining time: null
  Gerud Armorer:
    dropoff: null
    next discount: 10.123755634162158
    remaining time: null
  Gerud Forges:
    dropoff: null
    next discount: 57.61452534463683
    remaining time: null
  Gerud Harbor:
    dropoff: null
    next discount: 21.61452534463673
    remaining time: null
  Humil Dwarf Stable:
    dropoff: null
    next discount: 26.66266149812274
    remaining time: null
  James Blacksmith:
    dropoff: null
    next discount: 65.24736046426831
    remaining time: null
  Kathal Horses:
    dropoff: null
    next discount: 29.614525344636792
    remaining time: null
  Lorrys Grocery Store:
    dropoff: null
    next discount: 21.61452534463673
    remaining time: null
  Los Gerudos Grocery Store:
    dropoff: null
    next discount: 273.61452534463655
    remaining time: null
  Los Gerudos Hostel:
    dropoff: null
    next discount: 111.61452534463683
    remaining time: null
  Pickaxe Forge:
    dropoff: null
    next discount: 435.61452534463655
    remaining time: null
  Seaside Harbor:
    dropoff: null
    next discount: 39.61452534463681
    remaining time: null
  Seaside Hostels:
    dropoff: null
    next discount: 140.90454810063054
    remaining time: null
  Target Grocery:
    dropoff: 0.28
    next discount: -11.838375878665182
    remaining time: 0.3850337441127153
  Woodstack Hostel:
    dropoff: null
    next discount: 37.61452534463681
    remaining time: null
done missions:
- None
- Deliver Raw Wood To Jacky
- Cut Wood And Get Back Home
- Save Someone In The Woods
elapsed time game days: 151.18154344345461
elapsed time seconds: 90149.75679039955
enemies list:
- None
- Warg
- Dark Marksman
- Doomed Horror
- Cavern Troll
- Black Ork
- Ork Archer
- Slime
- Goblin
- Balrog
- White Warg
- Ork Archer
- Ork
- Corrupted Ent
- Woods Troll
- Mumakil
- Giant Spider
- South Pillager
- Desert Troll
- Shire Flying Beast
- Desert Undead
- Mangrove Undead
gold: 553.2111381263578
groceries data:
  Lorrys Grocery Store:
    date: 151
    items sales:
    - Dagger
    - Old Pants
    - Forlindon Woods Map
    - Chicken
    - Beef
    - Carrot
    - Hay Bale
    - Raw Salmon
  Los Gerudos Grocery Store:
    date: 151
    items sales:
    - Dagger
    - Strength Potion
    - Agility Potion
    - Wolf Fur
    - Tissue Bag
    - Gerud Desert Map
    - Carrot
    - Wheat
  Target Grocery:
    date: 151
    items sales:
    - Knight Beverage
    - Strength Potion
    - Wolf Fur
    - Old Pants
    - Broken Shield
    - Tissue Bag
    - Gerud Desert Map
    - Chicken
    - Beef
    - Wheat
    - Hay Bale
    - Raw Salmon
    - Cactus
    - Mumakil Tusk
    - Bow String
health: 117
heard dialogs:
- None
- 5
- 11
- 35
- 40
- 23
- 1035
- 1165
held boots: Knight Boots (2)
held chestplate: Wolf Fur (1)
held item: Flaming Whip
held leggings: Chainmail Leggings (4)
held shield: Warg Fur Shield
inventory:
- Rock Paper Scissors
- Magic Map
- Knight Boots (2)
- Stretched Bag
- Stretched Bag
- Stretched Bag
- Stretched Bag
- Stretched Bag
- Stretched Bag
- Stretched Bag
- Warg Fur Shield
- Ork Bag
- Ork Bag
- Ork Bag
- Tissue Bag
- Ork Bag
- Wolf Fur (1)
- Chainmail Leggings (4)
- Tissue Bag
- Tissue Bag
- Raw Wood
- Bow String
- Old Rusted Key
- Teleportation Rock
- Heart Stone
- Throwing Axe (2)
- Porkchop
- Beef
- Apple
- Beef
- Beef
- Forlindon Woods Map
- Stretched Bag
- Porkchop
- Gerud Desert Map
- Beef
- Knight Beverage
- Beef
- Beef
- Beef
- Stretched Bag
- Flaming Whip
- Strength Potion
- Resistance Potion
- Healing Potion
- Beef
- Porkchop
- Beef
- Strength Potion
- Healing Potion
- Bone
- Porkchop
- Beef
- Gerud Mercenary Beverage
- Resistance Potion
- Spear
- Warg Fur Shield
- Balrog Horn
inventory slots: 105
inventory slots remaining: 47
map zone: Goro Mounts Mines Dungeon
max health: 294
met npcs:
- None
- 35
met npcs names:
- Old Man
- Jerry The Junk Seller
- Marry
missions:
  Cut Wood And Get Back Home:
    stopovers went:
    - 1035
    - 1035
    went to all stopovers: true
  Deliver Raw Wood To Jacky:
    stopovers went: []
    went to all stopovers: false
  Mission 1:
    stopovers went: []
    went to all stopovers: false
  None: none
  Save Someone In The Woods:
    stopovers went: []
    went to all stopovers: false
mounts:
  03131d06-3b4a-483c-88f9-679b35a1c7db:
    deposited day: 9.51
    is deposited: true
    level: 0
    location: point6472
    mount: Camel
    mph: 10
    name: Cam The Camel
    stats:
      agility addition: 0.07
      resistance addition: 0.38
  1b78436f-cbab-46eb-8de5-5e5fdad338a7:
    deposited day: 536.2
    is deposited: true
    level: 2.4743846153846145
    location: point6472
    mount: Horse
    mph: 10
    name: Noireaude
    stats:
      agility addition: 0.115
      resistance addition: 0.235
  2aae8dad-930d-49b0-bcf3-1cb5c03b78dc:
    deposited day: 13.69
    is deposited: true
    level: 0
    location: point3499
    mount: Bear
    mph: 10
    name: Grizzly Friend
    stats:
      agility addition: -0.15
      resistance addition: 0.25
  3cd21029-36c6-41d8-8960-d2004b4bdb03:
    deposited day: 0
    is deposited: false
    level: 55
    location: point2856
    mount: White Warg
    mph: 39.8
    name: Mallow
    stats:
      agility addition: 1.08
      resistance addition: 0.78
  4e8dc1d4-c1b0-4357-8d5a-8b7b037d1c70:
    deposited day: 535.3
    is deposited: true
    level: 2.5019999999999993
    location: point6472
    mount: Friesian Horse
    mph: 10
    name: Silver Fries
    stats:
      agility addition: 0.172
      resistance addition: 0.324
  78281076-1336-4b1b-9e25-16bf8c6c9d49:
    deposited day: 10.5
    is deposited: true
    level: 0
    location: point6472
    mount: Donkey
    mph: 10
    name: Wild Ass
    stats:
      agility addition: 0.09
      resistance addition: 0.35
  bb553885-6eaa-4ac7-b36f-bda3b0862b47:
    deposited day: 535.9
    is deposited: true
    level: 1.841875
    location: point6472
    mount: Shire Horse
    mph: 10
    name: Pitro
    stats:
      agility addition: 0.155
      resistance addition: 0.33
offered missions:
- None
- Mission 1
- Deliver Raw Wood To Jacky
- Cut Wood And Get Back Home
- Save Someone In The Woods
orders:
  None: none
start dialog:
  dialog: start dialog
  heard start dialog: true
starting date: 04-12-1190
taken items:
- None
- 7
- 8
- 3
- 0
- 6
- 131
used keys:
- None
visited points:
- 1
- 2
- 3
- 4
- 5
- 6
- 7
- 8
- 9
- 10
- 11
- 13
- 12
- 14
- 15
- 16
- 17
- 18
- 19
- 20
- 21
- 22
- 23
- 24
- 25
- 26
- 27
- 28
- 29
- 30
- 31
- 32
- 33
- 34
- 35
- 36
- 37
- 42
- 43
- 38
- 39
- 40
- 0
- 100
- 59
- 60
- 61
- 62
- 63
- 64
- 66
- 65
- 67
- 69
- 70
- 71
- 72
- 73
- 74
- 75
- 76
- 77
- 78
- 79
- 80
- 86
- 85
- 84
- 51
- 48
- 87
- 89
- 104
- 119
- 132
- 133
- 134
- 121
- 106
- 105
- 98
- 99
- 101
- 2297
- 2296
- 2295
- 2294
- 2293
- 2292
- 2291
- 2290
- 2289
- 2288
- 2287
- 2286
- 2165
- 2044
- 2150
- 2204
- 2203
- 2202
- 2201
- 2200
- 2199
- 2198
- 2197
- 2326
- 2327
- 2328
- 2329
- 2330
- 2331
- 2332
- 2333
- 2334
- 2335
- 2336
- 2337
- 2338
- 2339
- 2340
- 2341
- 2342
- 2343
- 2344
- 2345
- 2346
- 2347
- 2348
- 2349
- 2478
- 2607
- 2736
- 2864
- 2992
- 3121
- 3250
- 3379
- 3508
- 3637
- 3766
- 3895
- 4024
- 4153
- 4282
- 4411
- 4540
- 4669
- 4798
- 4927
- 5056
- 5185
- 5314
- 5313
- 5312
- 5311
- 5310
- 5309
- 5308
- 5440
- 5569
- 5698
- 5827
- 5956
- 6085
- 6214
- 6343
- 6472
- 6473
- 6601
- 6730
- 6859
- 6988
- 6989
- 6990
- 6991
- 7119
- 6861
- 6732
- 7248
- 7377
- 7506
- 7635
- 7764
- 7763
- 7762
- 7761
- 7760
- 7759
- 7758
- 7757
- 7756
- 7755
- 7754
- 7753
- 7752
- 7751
- 7750
- 7621
- 7492
- 7363
- 7234
- 7105
- 6976
- 6847
- 6718
- 6589
- 6460
- 6331
- 6202
- 6073
- 5944
- 5815
- 5686
- 5557
- 5428
- 5299
- 5170
- 5041
- 4912
- 4783
- 4654
- 4525
- 4396
- 4267
- 4138
- 4009
- 3880
- 3751
- 3622
- 3493
- 3364
- 3235
- 3106
- 2978
- 2850
- 2721
- 2592
- 2463
- 2464
- 2465
- 2593
- 2462
- 2461
- 2460
- 2459
- 5945
- 5946
- 5947
- 5948
- 5949
- 5950
- 5951
- 5952
- 5953
- 5954
- 5955
- 5957
- 5958
- 5959
- 5960
- 5961
- 6084
- 6213
- 6342
- 6471
- 6600
- 6729
- 6731
- 6733
- 6734
- 6863
- 6992
- 7121
- 7250
- 7379
- 7508
- 7507
- 7765
- 7766
- 7767
- 7896
- 8025
- 8154
- 8283
- 8412
- 7638
- 7509
- 7505
- 7636
- 7894
- 8023
- 7895
- 7893
- 7378
- 7249
- 7120
- 7637
- 7510
- 7511
- 7512
- 7504
- 7503
- 7502
- 7501
- 7500
- 7499
- 7498
- 7497
- 7496
- 7495
- 7494
- 7493
- 7491
- 7490
- 7489
- 7488
- 7487
- 6605
- 6476
- 7358
- 7229
- 7100
- 6971
- 6842
- 6713
- 6584
- 6455
- 6326
- 6197
- 6068
- 5939
- 5810
- 5681
- 5552
- 5423
- 5294
- 5165
- 5036
- 4907
- 4778
- 4649
- 4520
- 4906
- 4905
- 5034
- 5163
- 5292
- 5421
- 5550
- 5551
- 5553
- 5554
- 5555
- 5556
- 5558
- 5559
- 5560
- 5561
- 5562
- 5563
- 5564
- 5565
- 5566
- 5567
- 5568
- 6470
- 6469
- 6468
- 6467
- 6466
- 6465
- 6464
- 6463
- 6462
- 6461
- 6459
- 6458
- 6457
- 6456
- 6454
- 6325
- 6196
- 6067
- 5938
- 5809
- 5680
- 5696
- 5825
- 6083
- 6212
- 6341
- 6599
- 6728
- 6857
- 6986
- 7115
- 7244
- 7373
- 7768
- 7513
- 7514
- 7515
- 7516
- 7517
- 7518
- 7519
- 7520
- 7521
- 7522
- 7523
- 6347
- 6218
- 6089
- 6088
- 6087
- 6086
- 6082
- 6081
- 6080
- 6079
- 6078
- 6077
- 6076
- 6075
- 6074
- 2205
- 2076
- 1947
- 1818
- 1689
- 1560
- 1431
- 1302
- 1173
- 1044
- 915
- 914
- 913
- 912
- 911
- 910
- 1039
- 1038
- 1037
- 1036
- 1035
- 1034
- 905
- 776
- 777
- 778
- 907
- 1165
- 1294
- 1295
- 1296
- 1425
- 1554
- 1293
- 1292
- 1421
- 1163
- 1033
- 906
- 1423
- 1552
- 1681
- 1810
- 1809
- 1680
- 1551
- 649
- 520
- 391
- 262
- 519
- 518
- 647
- 648
- 1683
- 1812
- 1941
- 2070
- 2196
- 2455
- 2584
- 2713
- 2842
- 2843
- 2970
- 2971
- 3098
- 3099
- 3227
- 3356
- 3485
- 3614
- 3743
- 3872
- 4001
- 4130
- 4259
- 4388
- 4517
- 4646
- 4775
- 4776
- 4777
- 5164
- 5162
- 5291
- 5293
- 5422
- 5570
- 5571
- 5699
- 5828
- 6215
- 6344
- 6602
- 6603
- 6474
- 7117
- 7118
- 7247
- 7246
- 7375
- 7374
- 7631
- 7630
- 7629
- 7632
- 7633
- 7634
- 8024
- 8153
- 8152
- 8151
- 8150
- 8149
- 8020
- 7891
- 7892
- 5831
- 5702
- 5573
- 5444
- 5315
- 5186
- 5057
- 4928
- 4799
- 4670
- 4541
- 4412
- 4283
- 4154
- 4025
- 5187
- 5188
- 5189
- 5184
- 5183
- 5182
- 5181
- 5439
- 5697
- 5826
- 6604
- 6345
- 6216
- 6217
- 6346
- 6475
- 6862
- 6864
- 6865
- 6866
- 904
- 1166
- 1164
- 1422
- 1682
- 1424
- 261
- 260
- 131
- 389
- 1297
- 1298
- 1299
- 1300
- 1301
- 1303
- 1304
- 1305
- 1306
- 1435
- 1564
- 1693
- 1822
- 1951
- 2080
- 2209
- 2467
- 2596
- 2725
- 2854
- 2982
- 3110
- 3239
- 3368
- 3497
- 3626
- 3755
- 3884
- 4013
- 4142
- 4271
- 4400
- 4529
- 4658
- 4787
- 4916
- 5045
- 5174
- 5303
- 5432
- 5690
- 5819
- 5820
- 5821
- 5822
- 5823
- 5824
- 5829
- 5830
- 5832
- 6090
- 6219
- 6348
- 6477
- 6860
- 7376
- 7122
- 7123
- 8022
- 6858
- 6993
- 6987
- 6985
- 6984
- 6983
- 6982
- 6981
- 6980
- 6979
- 6978
- 6977
- 6975
- 6974
- 6845
- 6716
- 6587
- 6329
- 6200
- 6071
- 5942
- 5813
- 5684
- 5426
- 5297
- 5168
- 5039
- 4910
- 4781
- 4652
- 4523
- 4394
- 4265
- 4136
- 4007
- 3878
- 3749
- 3620
- 3619
- 3618
- 3617
- 3616
- 3615
- 2583
- 2454
- 2325
- 2067
- 2066
- 2065
- 2064
- 2068
- 2069
- 2071
- 2072
- 1943
- 1814
- 1685
- 1556
- 1427
- 1426
- 390
- 2457
- 2586
- 2715
- 2844
- 2972
- 3100
- 3229
- 3358
- 3487
- 3745
- 3874
- 4003
- 4132
- 4261
- 4390
- 4519
- 4648
- 5035
- 3499
- 2591
- 3621
- 3747
- 3746
- 5037
- 5038
- 5040
- 6719
- 6720
- 6721
- 6722
- 6723
- 6724
- 6725
- 6726
- 6727
- 6994
- 6995
- 6996
- 6997
- 6998
- 6999
- 7000
- 2973
- 2974
- 2975
- 2976
- 2848
- 2719
- 2590
- 2075
- 2077
- 2078
- 2079
- 3498
- 3628
- 3757
- 3886
- 4015
- 4144
- 4273
- 4402
- 4531
- 4660
- 4789
- 4918
- 5047
- 5176
- 5305
- 5434
- 5433
- 5431
- 5430
- 5429
- 5427
- 5425
- 5424
- 6855
- 6856
- 7113
- 7242
- 7371
- 7114
- 7116
- 7124
- 7125
- 7380
- 7381
- 4779
- 4780
- 4391
- 4389
- 3873
- 3875
- 3876
- 3877
- 3879
- 3744
- 3486
- 3357
- 3228
- 3230
- 3231
- 3232
- 3233
- 3234
- 2979
- 2980
- 3108
- 3237
- 3366
- 3367
- 3500
- 3501
- 3502
- 3503
- 3504
- 3505
- 3634
- 3763
- 3892
- 4021
- 4150
- 4279
- 4408
- 4409
- 4410
- 4800
- 4930
- 4929
- 5441
- 646
- 5055
- 4797
- 4796
- 4795
- 4794
- 4793
- 4792
- 4791
- 4790
- 4788
- 4786
- 4661
- 4532
- 4403
- 4274
- 4145
- 4016
- 3887
- 3758
- 3629
- 3496
- 3495
- 3494
- 3492
- 3491
- 3362
- 2585
- 1179
- 6598
- 6340
- 6211
- 2714
- 5033
- 4904
- 5166
- 4262
- 4133
- 4004
- 3488
- 3359
- 3101
- 2845
- 2716
- 2587
- 2588
- 2589
- 2074
- 2073
- 2456
- 2458
- 1180
- 1308
- 1437
- 1566
- 1695
- 1824
- 1825
- 1954
- 2083
- 2084
- 2082
- 2211
- 2469
- 2598
- 2727
- 2856
visited zones:
- Stall Fields
- Cocorico Village
- Woodstack Hostel
- Clarc Warg Stable
- James Blacksmith
- Pickaxe Forge
- The Edge Of Goro Mounts
- The Goro Mounts
- The Goro Mounts Peaks
- Californ Woods
- Seaside Hostels
- Gerud Lakes
- Gerud Desert
- Badlands
- Gerud Valleys
- Kathal Horses
- Los Gerudos
- Los Gerudos Hostel
- Gerud Forges
- Middle Area Lakes
- Mangrove Wastes
- Stall Seas Bay
- Gerud Armorer
- Stall Seas
- Magrove Wastes Waters
- New Hyru
- Forlindon Woods
- Ayar Town
- Forlindon Lake
- Californ Beaches
- Humil Dwarf Stable
- Cocorico Church
- Hyru Church
- Los Gerudos Church
- Los Gerudos Grocery Store
- Lorry's Grocery Store
- Target Grocery
- Gerud Harbor
- Seaside Harbor
- Dwarf Harbor
- Goro Mounts Mines Dungeon
walked miles: 1091
x: -46
xp: 13437.029999999833
y: 42
```

</details>

Don't forget to manually chance your game data at `<game dir>/game/`, so it has the PR updated data.
Run the `Z` command and enter the dungeon to test it! You can also change the data in `zone.yaml` to test stuff like other enigmas or other parameters.

For the reason of making some enigmas work, a new variable was added to the [game re-usable variables](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/wiki/Game-Re%E2%80%90Usable-Variables): the `preferences` dictionary.

## Performance Impact
N/A

## Check-List
- [x] Code basic implementation
- [x] Code dungeon UI
- [x] Create basic enigmas templates in `scripts/`
- [x] Add different dungeons across the map
- [x] Update submodules
- [x] Update BOW-Progression repo
- [x] Update wiki page about map zones to include the dungeon type (https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/wiki/Creating-Map-Zones#dungeons)
- [x] Fix checks

### CODE REVIEW

**Review by OcelotWalrus — COMMENTED** [review]
Hey @lumbar527! Would you mind giving a small review to this PR?

File: source/zone_handling.py [hunk 1]
```diff
@@ -1497,11 +1500,70 @@ def interaction_harbor(map_zone, zone, map, player):
                     "INFO: Canceling ticket buying process --> current harbor" +
                     f"doesn't have a ticket named '{which_ticket}'"
                 )
-                cout(COLOR_YELLOW + "There isn't any ticket name like that" + COLOR_RESET_ALL)
+                cout(COLOR_YELLOW + "There isn't any ticket named like that" + COLOR_RESET_ALL)
```
Comment (lumbar527) [inline source/zone_handling.py:1503]:
```suggestion
                cout(COLOR_YELLOW + "There isn't any ticket named that" + COLOR_RESET_ALL)
```

**Review by lumbar527 — APPROVED** [review]
Looks cool but I haven't tested!

**Comment by OcelotWalrus** [comment]
> Looks cool but I haven't tested!

I should probably add screenshots to the PR description. I also have to work on the [Gameplay Guide](https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs/wiki/Gameplay-Guide) dungeon section.

**Pull request merged**